common/SC_Win32Utils.cpp

```diff
@@ -52,12 +52,18 @@ void win32_ExtractContainingFolder(char* folder, const char* pattern, int maxCha
         folder[0] = 0;
 }
 
-void win32_GetKnownFolderPath(int folderId, char* dest, int size) {
-    // Use a temporary buffer, as SHGetFolderLocation() requires it
-    // to be at least MAX_PATH size, but destination size may be less
-    char buf[MAX_PATH];
-    SHGetFolderPath(NULL, folderId, NULL, 0, buf);
-    strncpy(dest, buf, size);
+std::wstring win32_StringToWstring(const std::string& string) {
+    DWORD size = MultiByteToWideChar(CP_UTF8, 0, string.c_str(), -1, nullptr, 0);
+    std::wstring outputString(size, 0);
```

Comment:
Similarly, I'm checking whether size is > 0. However, that is done elsewhere in the Env var code, typically.

Should I do the rest like
```cpp
if(size > 0) {
    std::wstring outputString(size, 0);
    MultiByteToWideChar(CP_UTF8, 0, string.c_str(), -1, outputString.data(), string.length());
    return outputString;
} else {
   return nullptr; // or an empty string?
}
```